### ISSUE TITLE
K9 Android - Cannot properly parse comma-separated email addresses when pasting list of addresses to TO and/or CC fields.

### ISSUE THREAD

**Issue by hammerjammer** [thunderbird/thunderbird-android#2961]
### Expected behavior
Copying a list of comma-separated email addresses from any plaintext editor into the TO and/or CC fields if K9 in Android should result in a properly displayed and parsed list of email addresses like every other normal email client application.

### Actual behavior
K9 under Android totally mangles the list except for either the first or last email address. Mangled results range from extra . inserted or , not recognized or a +1 string added BETWEEN email addresses or a K9 error popup with a red exclamation mark stating "Error parsing address".

### Steps to reproduce
1.create a list of email addresses you want to send your email to inside a plaintext editor UNDER ANDROID. 
For example:  email1@test.com, email2@screwedup.com, testing@me.com, screwk9@kshit.com

2.copy and paste the above list into any TO or CC field in K9 under Android.

3. Tap any other field to leave to TO or CC field you pasted the email list to.

4. Watch K9 add a +1 between the first and remaining email addresses. Then when you tap on the TO or CC field to try to fix the mangling, K9 mangles it further and in some cases a red exclamation mark comes up saying Error Parsing Address.

### Environment
K-9 Mail version: 5.208
Android version: 7.1.1
Account type (IMAP, POP3, WebDAV/Exchange): POP3

**Comment by Olf0** [comment]
+1: This behaviour introduced with K9 5.2 is really annoying (K9 5.0 handled this well).

**Comment by ma651851384** [comment]
> This behaviour introduced with K9 5.2 is really annoying (K9 5.0 handled this well).

I totally agree! I don't know the version of faulty K9 version but multiple email adresses in recipient field does not work anymore. I was using that functionality very often but now its not possible anymore. I am suprised because it seems that nearly no one used k9 in that way.

Please make it possible again to use multiple addresses as descripted from hammerjammer.

**Comment by ufdemt15** [comment]
I've recently had the same issue trying to copy email address from one email to another. I had to do this because just hitting reply all with a large number, more than 2 or 3, will not send right and fails.

**Comment by dkbg** [comment]
I've come across a similar issue with K-9 5.403. When pasting addresses, they don't get mangled but I get a parsing error and the addresses don't appear to be properly detected (doesn't convert into a little rectangular pill). Only workaround I can find is to copy and paste each address separately, typing a comma after each so it's detected.

**Comment by francwalter** [comment]
I use the workaround with the App "Group Share", but this is not the best solution. Would like to see this fixed.
Thanks.

**Comment by ma651851384** [comment]
Sorry, but meantime I switched to another android email app.

**Comment by francwalter** [comment]
> Sorry, but meantime I switched to another android email app.

😄  me too, me too 👍 
FairEmail is mine, since a while already and never wanted back 😏

**Comment by numericOverflow** [comment]
Here's a screen capture of me trying multiple combinations of CRLF, semicolon, and comma, none of which allow me to paste multiple addresses into the "TO/CC/BCC" field.

You really need to make this feature work, it's is a *MUST HAVE* 

https://user-images.githubusercontent.com/3429424/166165871-aedf59d7-2570-4ff2-886d-d19dd2bcaeff.mp4

!!!

**Comment by numericOverflow** [comment]
So will anyone be fixing this bug, or will it just stagnate indefinitely?

**Comment by cketti** [comment]
@numericOverflow: We do accept pull requests to fix this bug.

**Comment by numericOverflow** [comment]
@cketti - Unfortunately I don't have much (or any) experience in android programming, nor do I have a build environment to help out.

**Comment by Olf0** [comment]
@numericOverflow, this is not how Free Software development works, which is not backed by a company you are a paying customer of.

> You really need to make this feature work, it's is a MUST HAVE

Please do understand that you are not in a position to tell anybody here, what needs to be done.  And shouting at people (all caps) is even more detrimental to motivate someone who already has far enough tasks piled up to accept another one.

If you really think so, work on this feature.

> So will anyone be fixing this bug, or will it just stagnate indefinitely?

Here "anyone" points to nobody (hence the terms are equal in this context), hence it may be pending indefinitely.

> Unfortunately I don't have much (or any) experience in android programming, nor do I have a build environment to help out.

Well, then either get educated (all necessary info is in the WWW) and install an build environment or leave it.
In my view this an often heard and obvious try to bail out cheaply.

**Comment by numericOverflow** [comment]
@Olf0 - I understand how free/open source software development works and I fully appreciate all the hard work any/all developers do.  I contribute to projects whenever I can, I'm just not a very capable java developer and haven't contributed to anything on android.

That said, it was not my intention to yell at or belittle anyone over this bug, and I definitely agree reading back that my comments did seem pompous/entitled.  

I apologize for my harsh tone.  I was just very sad to see this bug had been around since 2017 and I shouldn't have taken it out on any of the hard working developers that make K9 mail a great app.

I poked around the source code briefly and there seems to be a function that's meant to parse addresses, it just likely not working as expect or getting fired properly.  I'll try to dig in more to provide whatever help I can.

Again, my sincerest apologies for offending anyone
